### PR TITLE
doc: Pairing

### PR DESCRIPTION
**Pull request by jdjkelly**
## Why
* Flexpa uses paired programming as a regular development technique and we've evolved a set of guidelines on how it ought to work (in particular for remote)

## What
* Adds a new Pairing guide to Processes, re-written from an internal draft / feedback cycle

### CODE REVIEW

File: docs/process/pairing.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+Along all three axes, personal preferences and performance factors should be communicated.
+If you prefer unblocking problems as a navigator in ad hoc pairs, you should make sure to say it. 
+
+## How
+
+Pairing requires active engagement by both people.
+There is no passive role.
+
+Typically, at least when in person, pairing means sharing one computer to do the work together.
+Working on separate machines is considered bad etiquette at best and at worst missing the point entirely.
```
Comment (dylman123) [inline docs/process/pairing.md:62]: I was surprised to read this. In person, you (@jdjkelly) and I paired on separate machines and I didn't perceive it as bad etiquette. In fact I thought pairing on the same machine may have limited the amount of screen/monitor real estate that we could use and may have hindered our ability to move fast. What do you think?

**Review by dylman123 — CHANGES_REQUESTED** [review]
Looks good, left a couple comments

File: docs/process/pairing.md (hunk 1 above)
Comment (jdjkelly) [reply docs/process/pairing.md:62]: You may not have noticed but I sat away from my laptop, and at times closed it entirely. I didn't use my machine throughout the pair except when I announced that I was doing something. When we needed to use one of my accounts, we actually didn't use my machine but instead screen shared it onto yours. All of that was intentional for the reasons I write here.

File: docs/process/pairing.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+This is a trap that is easy to fall into as a navigator and must be consciously avoided.
+
+**Scheduled** pairs are straightforward: you agree to a time in advance.
+**Ad hoc** pairs happen immediately.
+You must actively communicate personal preferences and performance factors when being asked to pair. 
+
+Do not accept invites or requests to pair if you _know_ you are not going to be an active, great communicator.
+Consider asking “is now still a good time to pair?” when you begin.
+The only pair you are obliged to accept is one with the CTO.
+That said, if you find yourself never offering or accepting a pair don’t be surprised to hear feedback about teamwork.
```
[the remark this answers is not in the document]
Comment (jdjkelly) [reply docs/process/pairing.md]: I chose to remove this content, and let the statement be briefer than overly-specified. We have to be careful of over-specification.

**Review by amjarruda — APPROVED** [review]
Great update. Very clear.

**Pull request merged**